app/assets/javascripts/provider/cms_toolbar_v2.js

```diff
@@ -0,0 +1,51 @@
+// = require 'vendor/jquery-3.5.0.min.js'
+// = require 'vendor/jquery/extensions/jquery.cookie.js'
+
+$(function () {
+  const toolbar = $('#cms-toolbar')
+  const iframe = $('#developer-portal')
+  const toolbarMode = $('form#cms-toolbar-mode')
+  toolbarMode.find('li').on('click', () => {
+    $(this).closest('li').find('input').attr('checked', true)
+    $(toolbarMode).trigger('change')
+  })
+  toolbarMode.on('change', () => {
+    window.location = $(this).find('input:checked').val()
+  })
+  const enableAnimation = () => {
+    toolbar.addClass('animate')
+    iframe.addClass('animate')
+  }
+  const toggleValues = () => {
+    toolbar.toggleClass('not-hidden')
+    iframe.toggleClass('not-full')
+  }
+  const storedToolbarState = () => {
+    $.cookie('cms-toolbar-state', {
+      path: '/'
+    })
+  }
+  const saveToolbarState = state => {
+    $.cookie('cms-toolbar-state', state, {
+      expires: 30,
+      path: '/'
+    })
+  }
+  iframe.on('load', () => {
+    if (storedToolbarState() !== 'hidden') {
+      toggleValues()
+      return (window.requestAnimationFrame || window.setTimeout)(enableAnimation)
+    } else {
+      return enableAnimation()
+    }
+  })
+  $('#hide-side-bar').on('click', event => {
+    event.preventDefault()
+    toggleValues()
+    if (storedToolbarState() === 'hidden') {
+      saveToolbarState('visible')
+    } else {
+      saveToolbarState('hidden')
+    }
```

Comment:
Should we modernize the code as well?
```suggestion
    saveToolbarState(storedToolbarState() === 'hidden' ? 'visible' : 'hidden')
```

Reply:
But then why "modernizing" this line and not the others? The whole file can be "modernized", but I'd do it as a separate task.

Reply:
This was only an example, but yes let's leave it for later!